Walk

L98293 opened the pull request:
산책구현, 쿠키 최적화

L98293 committed:
feat: 산책시작, 중단, 완료 구현

L98293 committed:
feat: 프로필 조회 구현

L98293 committed:
fix: 주석 추가

L98293 committed:
feat: 토큰 검증 추가

L98293 committed:
feat: 랭킹 반환

JPQL을 사용하여 산책거리순으로 내림차순 정렬 후 반환 및 로그아웃 시 쿠키 삭제되도록 수정

L98293 commented on src/main/java/com/GoWalk/domain/member/application/usecase/TokenUseCase.java:
```diff
@@ -55,52 +55,42 @@ public String generateRefreshToken(GenerateTokenReq tokenReq, String userId, Htt
+
+			Cookie accessCookie = new Cookie("accessToken", newAccessToken);
+			accessCookie.setPath("/");
+			accessCookie.setHttpOnly(false);
```
? 그럼 만료된걸 어케 알수 있져? 
일단 보류

L98293 committed:
feat: 이메일 인증 추가

daehyeong2 commented on src/main/java/com/GoWalk/domain/member/application/data/req/SignUpInReq.java:
```diff
@@ -7,8 +7,7 @@ public record SignUpInReq(
-		String refreshToken,
-		String accessToken
+		Integer breedAge,
+		Double walkDistance
```
이건 따로 req로 받지도 않고, entity에도 저장 안합니다. 이건 산책 세션 join해서 sum() 하는거로 구할 예정입니다
(walkDistance)

L98293 committed:
feat: 토큰 저장 방식 변경, 랭킹 추가

1. 기존 userId로 분류했던 토큰을 email로 분류되도록 변경했습니다.
2. 로그아웃 시 값을 요청받는 걸 토큰에서 가져오도록 변경하였습니다.
3. 엑세스 토큰 재발급의 경우도 리프레시 토큰을 활용하도록 변경하였습니다.
4. 산책일수로 랭킹을 조회할 수 있도록 변경했습니다.
5. 불필요한 코드들을 삭제하였습니다.

daehyeong2 reviewed (approved):
머지하면 될 것 같습니다.
